types/imagekit/index.d.ts

```diff
@@ -6,929 +6,948 @@
+        }
+
+        /**
+         * Response when purging CDN and ImageKit.io internal cache
+         *
+         * @see {@link https://docs.imagekit.io/api-reference/media-api/purge-cache#response-structure-and-status-code}
+         */
+        interface PurgeCacheResponse {
+            /**
+             * requestId can be used to fetch the status of submitted purge request.
+             */
+            requestId: string;
+        }
+
+        /**
+         * Response when getting the status of submitted purge request.
+         *
+         * @see {@link https://docs.imagekit.io/api-reference/media-api/purge-cache-status#understanding-response}
+         */
+        interface PurgeCacheStatusResponse {
+            /**
+             * Pending - The request has been successfully submitted, and purging is in progress.
+             * Complete - The purge request has been successfully completed. And now you should get a fresh object.
+             * Check the Age header in response to confirm this.
+             */
+            status: 'Pending' | 'Completed';
+        }
+
+        interface Callback<T, E extends Error = Error> {
+            (error?: E, response?: T): void;
+        }
+    }
+
+    class ImageKit {
+        constructor(options: {
+            publicKey: string;
+            privateKey: string;
+            urlEndpoint: string;
+            transformationPosition?: ImageKit.TransformationPosition;
+        });
+
+        /**
+         * You can add multiple origins in the same ImageKit.io account.
+         * URL endpoints allow you to configure which origins are accessible through your account and set their preference order as well.
+         *
+         * @see {@link https://github.com/imagekit-developer/imagekit-nodejs#url-generation}
+         * @see {@link https://docs.imagekit.io/integration/url-endpoints}
+         *
+         * @param urlOptions
+         */
+        url(urlOptions: ImageKit.UrlOptions): string;
+
+        /**
+         * This API can list all the uploaded files in your ImageKit.io media library.
+         * For searching and filtering, you can use query parameters as described below.
+         *
+         * @see {@link https://docs.imagekit.io/api-reference/media-api/list-and-search-files}
+         *
+         * @param listFilesOptions
+         * @param callback
+         */
+        listFiles(
+            listFilesOptions: ImageKit.ListFileOptions,
+            callback: ImageKit.Callback<ImageKit.ListFileResponse>,
+        ): void;
+
+        /**
+         * This API can list all the uploaded files in your ImageKit.io media library.
+         * For searching and filtering, you can use query parameters as described below.
+         *
+         * @see {@link https://docs.imagekit.io/api-reference/media-api/list-and-search-files}
+         *
+         * @param listFilesOptions
+         */
+        listFiles(listFilesOptions: ImageKit.ListFileOptions): Promise<ImageKit.ListFileResponse>;
```

Comment:
It looks like this should be `Promise<ImageKit.ListFileResponse[]>`

https://docs.imagekit.io/api-reference/media-api/list-and-search-files

`The JSON-encoded response has an array of items. Each item can have the following properties.`